### PR TITLE
Add a test that ensures we are using SubtleCrypto in WASM crypto tests

### LINKED ISSUE #69740: WASM error handling in SubtleCrypto through web worker
The work to enable WASM to use SubtleCrypto, https://github.com/dotnet/runtime/pull/65966, had some follow ups on how to handle error cases. Specifically, the communication channel with the web worker could be hardened to convey more errors/failures. This could be accomplished through throwing an exception or the web worker communicating a failure case explicitly.

Source hints:
- `crypto-worker.ts`
    - see `read_response` and consider throwing an `error`.
- `dotnet-crypto-worker.js`
    - The reading in of a request could be including in the `try`/`catch` in `await_request`.
    - The `await_request` function could return a bespoke "error occurred" message instead of an exception string.

See https://github.com/dotnet/runtime/issues/69741

/cc @radical @ericstj @layomia

### PR DESCRIPTION
**Pull request by eerhardt**
Fix #69740

### CODE REVIEW

**Comment by bartonjs** [comment]
The C# LGTM.  The rest also LGTM, but in more of a nod-and-smile sort of way.

File: src/libraries/System.Security.Cryptography/tests/wasm.helix.targets [hunk 1]
```diff
@@ -21,8 +21,12 @@
...
         <Timeout>$(_workItemTimeout)</Timeout>
 
-        <PreCommands Condition="'$(OS)' == 'Windows_NT'">set &quot;WasmXHarnessArgs=%WasmXHarnessArgs% --web-server-use-cop&quot;</PreCommands>
-        <PreCommands Condition="'$(OS)' != 'Windows_NT'">export &quot;WasmXHarnessArgs=$WasmXHarnessArgs --web-server-use-cop&quot;</PreCommands>
+        <PreCommands Condition="'$(OS)' == 'Windows_NT'">set &quot;WasmXHarnessArgs=%WasmXHarnessArgs% --web-server-use-cop&quot; &amp; set &quot;WasmXHarnessMonoArgs=--setenv=TEST_EXPECT_SUBTLE_CRYPTO=true&quot;</PreCommands>
+        <PreCommands Condition="'$(OS)' != 'Windows_NT'">export &quot;WasmXHarnessArgs=$WasmXHarnessArgs --web-server-use-cop&quot; &amp;&amp; export &quot;WasmXHarnessMonoArgs=--setenv=TEST_EXPECT_SUBTLE_CRYPTO=true&quot;</PreCommands>
+       </HelixWorkItem>
+ 
+       <_CryptoHelixItem Include="@(HelixWorkItem)"
+
```
Comment (radical) [inline src/libraries/System.Security.Cryptography/tests/wasm.helix.targets:29]:
```suggestion
```

Comment (radical) [inline src/libraries/System.Security.Cryptography/tests/wasm.helix.targets:29]:
```suggestion
```

File: src/libraries/System.Security.Cryptography/tests/wasm.helix.targets [hunk 2]
```diff
@@ -21,8 +21,12 @@
         <Command>$(HelixCommand)</Command>
         <Timeout>$(_workItemTimeout)</Timeout>
 
-        <PreCommands Condition="'$(OS)' == 'Windows_NT'">set &quot;WasmXHarnessArgs=%WasmXHarnessArgs% --web-server-use-cop&quot;</PreCommands>
-        <PreCommands Condition="'$(OS)' != 'Windows_NT'">export &quot;WasmXHarnessArgs=$WasmXHarnessArgs --web-server-use-cop&quot;</PreCommands>
+        <PreCommands Condition="'$(OS)' == 'Windows_NT'">set &quot;WasmXHarnessArgs=%WasmXHarnessArgs% --web-server-use-cop&quot; &amp; set &quot;WasmXHarnessMonoArgs=--setenv=TEST_EXPECT_SUBTLE_CRYPTO=true&quot;</PreCommands>
+        <PreCommands Condition="'$(OS)' != 'Windows_NT'">export &quot;WasmXHarnessArgs=$WasmXHarnessArgs --web-server-use-cop&quot; &amp;&amp; export &quot;WasmXHarnessMonoArgs=--setenv=TEST_EXPECT_SUBTLE_CRYPTO=true&quot;</PreCommands>
```
Comment (radical) [inline src/libraries/System.Security.Cryptography/tests/wasm.helix.targets:25]:
```suggestion
        <PreCommands Condition="'$(OS)' != 'Windows_NT'">export &quot;WasmXHarnessArgs=$WasmXHarnessArgs &quot;--web-server-use-cop&quot; &amp;&amp; export &quot;WasmXHarnessMonoArgs=--setenv=TEST_EXPECT_SUBTLE_CRYPTO=true&quot;</PreCommands>
```

File: src/libraries/System.Security.Cryptography/tests/wasm.helix.targets [hunk 3]
```diff
@@ -21,8 +21,8 @@
         <Command>$(HelixCommand)</Command>
         <Timeout>$(_workItemTimeout)</Timeout>
 
-        <PreCommands Condition="'$(OS)' == 'Windows_NT'">set &quot;WasmXHarnessArgs=%WasmXHarnessArgs% --web-server-use-cop&quot;</PreCommands>
-        <PreCommands Condition="'$(OS)' != 'Windows_NT'">export &quot;WasmXHarnessArgs=$WasmXHarnessArgs --web-server-use-cop&quot;</PreCommands>
+        <PreCommands Condition="'$(OS)' == 'Windows_NT'">set &quot;WasmXHarnessArgs=%WasmXHarnessArgs% --web-server-use-cop&quot; &amp; set &quot;WasmXHarnessMonoArgs=--setenv=TEST_EXPECT_SUBTLE_CRYPTO=true&quot;</PreCommands>
```
Comment (radical) [inline src/libraries/System.Security.Cryptography/tests/wasm.helix.targets:24]:
```suggestion
        <PreCommands Condition="'$(OS)' == 'Windows_NT'">set &quot;WasmXHarnessArgs=%WasmXHarnessArgs% &quot;--web-server-use-cop&quot; &amp; set &quot;WasmXHarnessMonoArgs=--setenv=TEST_EXPECT_SUBTLE_CRYPTO=true&quot;</PreCommands>
```

File: src/libraries/System.Security.Cryptography/tests/System.Security.Cryptography.Tests.csproj [hunk 4]
```diff
@@ -9,8 +9,12 @@
...
   <PropertyGroup>
     <TargetPlatformIdentifier>$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)'))</TargetPlatformIdentifier>
-
-    <HelixTargetsFile Condition="'$(TargetOS)' == 'Browser'">wasm.helix.targets</HelixTargetsFile>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetOS)' == 'Browser'">
+    <HelixTargetsFile>wasm.helix.targets</HelixTargetsFile>
+ 
+    <WasmXHarnessMonoArgs Condition="'$(ContinuousIntegrationBuild)' != 'true' and '$(UseSubtleCryptoForTests)' == 'true'">$(WasmXHarnessMonoArgs) --setenv=TEST_EXPECT_SUBTLE_CRYPTO=true</WasmXHarnessMonoArgs>
+    <WasmXHarnessArgs Condition="'$(ContinuousIntegrationBuild)' != 'true' and '$(UseSubtleCryptoForTests)' == 'true'">$(WasmXHarnessArgs) --web-server-use-cop&quot;</WasmXHarnessArgs>
```
Comment (radical) [inline src/libraries/System.Security.Cryptography/tests/System.Security.Cryptography.Tests.csproj:17]:
```suggestion
    <WasmXHarnessArgs Condition="'$(ContinuousIntegrationBuild)' != 'true' and '$(UseSubtleCryptoForTests)' == 'true'">$(WasmXHarnessArgs) &quot;--web-server-use-cop&quot;</WasmXHarnessArgs>
```

**Review by radical — APPROVED** [review]

**Comment by radical** [comment]
Wasm failures are known, and unrelated to this PR.

**Comment by radical** [comment]
Failures are https://github.com/dotnet/runtime/issues/72912, and https://github.com/dotnet/runtime/issues/72916 .

**Pull request merged**